test/e2e/v4/plugin_cluster_test.go

```diff
@@ -185,25 +185,25 @@ func Run(kbc *utils.TestContext, hasWebhook, isToUseInstaller, isToUseHelmChart,
 		"pod", controllerPodName,
 		"-o", "jsonpath={.spec.containers[0].args}",
 	)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred())
-	ExpectWithOffset(1, podOutput).To(ContainSubstring("leader-elect"),
+	Expect(err).NotTo(HaveOccurred())
+	Expect(podOutput).To(ContainSubstring("leader-elect"),
 		"Expected manager pod to have --leader-elect flag")
-	ExpectWithOffset(1, podOutput).To(ContainSubstring("health-probe-bind-address"),
+	Expect(podOutput).To(ContainSubstring("health-probe-bind-address"),
 		"Expected manager pod to have --health-probe-bind-address flag")
 
 	By("validating that the Prometheus manager has provisioned the Service")
-	EventuallyWithOffset(1, func() error {
+	Eventually(func(g Gomega) {
```

Comment:
I think there is no need to change function inside Eventually. The `.Should(Succeed())` will do the assertion for us

Reply:
I found it a bit simpler to not have return values inside these functions. It's a bit simpler to reason as there is nothing returned.  

That way we can already immediately see that the error should never occur. Whereas when we return it it could be still checked for `NotFound` etc.

Reply:
Using a func(Gomega) in an Eventually is more "scalable" in that if you want to do more things inside the function, you can keep to the Gomega semantics with assertions.

* The error messages tell you exactly what's wrong.
* The line number of the error is correct (the error points you to the line inside the func() not just the line of the Eventually() call).
* No need to switch between two different ways of checking for errors in tests.

So my vote is to keep this change.